### PR TITLE
Kyle/filter and search

### PR DESCRIPTION
**Pull request by kyleqhua**
## Filter and Search for Studio Assessments

Added filtering and search abilities for the dashboard. Fixed databases models as well for Studio Assessments and Users.

### Screenshots
What I made:
Screen Shot 2020-04-15 at 11 02 49 PM

Goal:
Screen Shot 2020-04-15 at 11 14 54 PM

CC: @didvi

### CODE REVIEW

File: app/controllers/studio_assessments_controller.rb [hunk 1]
```diff
@@ -5,6 +5,24 @@ class StudioAssessmentsController < ApplicationController
...
+        "name" => s.participant.full_name,
+        "bigpicture_score" => s.bigpicture_score,
+        "progfundamentals_score" => s.progfundamentals_score,
+        "versioncontrol_score" => s.versioncontrol_score,
+        "react_score" => s.react_score,
+        "node_score" => s.node_score,
+        "db_score" =>s.db_score,
+        "problemsolving_score" => s.problemsolving_score,
+        "problemsolvingalt_score" => s.problemsolvingalt_score,
+        "id" => s.participant_id,
```
Comment (didvi) [inline app/controllers/studio_assessments_controller.rb:20]: is it necessary to create new objects? can we just use StudioAssessment.all ?

File: app/views/assignments/index.html.erb [hunk 2]
```diff
@@ -1,3 +1,5 @@
+
+<%= react_component("ActionItemDashboard", { is_admin: @user.admin?}) %>
```
Comment (didvi) [inline app/views/assignments/index.html.erb:2]: change to `current_user.admin?`

File: app/views/studio_assessments/index.html.erb [hunk 3]
```diff
@@ -66,3 +67,6 @@
 <br>
 
 <%= link_to 'New Studio Assessment', new_studio_assessment_path %>
+<% end %>
+
+<%= react_component("StudioAssessmentDashboard", {assessments: @studio_list,  is_admin: @user.admin? }) %>
```
Comment (didvi) [inline app/views/studio_assessments/index.html.erb:72]: `current_user.admin?`

File: app/controllers/assignments_controller.rb [hunk 4]
```diff
@@ -2,6 +2,7 @@ class AssignmentsController < ApplicationController
     before_action :set_action_item, only:[:show, :edit]
     def index
         @assignments = authorize Assignment.all
+        @user = current_user
```
Comment (didvi) [inline app/controllers/assignments_controller.rb:5]: rm

File: app/views/assignments/index.html.erb (hunk 2 above)
Comment (didvi) [reply app/views/assignments/index.html.erb:2]: also remove unused html

File: app/views/studio_assessments/index.html.erb [hunk 5]
```diff
@@ -1,3 +1,4 @@
+<% if false %>
 <p id="notice"><%= notice %></p>
```
Comment (didvi) [inline app/views/studio_assessments/index.html.erb:3]: remove all the unused html

**Comment by didvi** [comment]
Can you also add a column that displays the date that the studio assessment was last edited

File: app/controllers/studio_assessments_controller.rb (hunk 1 above)
Comment (kyleqhua) [reply app/controllers/studio_assessments_controller.rb:20]: nah i was just playing around with some stuffs

Comment (kyleqhua) [reply app/controllers/studio_assessments_controller.rb:20]: wait i remeber why I made it like that. To get the name I used participant.full_name, which is a function that gets the full name.

File: app/javascript/components/StudioAssessmentDashboard/index.js [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+    const assessments = this.state.trie.get(searchVal);
+    this.setState({
+      assessments,
+    });
+  }
+
+  handleDropdown = (event) => {
+    const category = event.target.value;
+    let currAssessments = this.props.assessments;
+    if(category === "overall"){
```
[the remark this answers is not in the document]
Comment (kyleqhua) [reply app/javascript/components/StudioAssessmentDashboard/index.js:58]: this is big brain TY

**Pull request merged**